Mathlib/Topology/Compactness/PseudometrizableLindelof.lean

```diff
@@ -0,0 +1,38 @@
+/-
+Copyright (c) 2023 Josha Dekker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Josha Dekker
+-/
+
+import Mathlib.Topology.Metrizable.Basic
+import Mathlib.Topology.Compactness.Lindelof
+
+/-!
+# Second-countability of pseduometrizable Lindelöf spaces
+
+Factored out from `Mathlib.Topology.Compactness.Lindelof`
+to avoid circular dependencies.
+-/
+
+variable {X : Type*} [TopologicalSpace X]
+
+open Set Filter Topology TopologicalSpace
+
+instance SecondCountableTopology.ofPseudoMetrizableSpaceLindelofSpace [PseudoMetrizableSpace X]
+    [LindelofSpace X] : SecondCountableTopology X := by
+  letI : PseudoMetricSpace X := TopologicalSpace.pseudoMetrizableSpacePseudoMetric X
+  have h_dense : ∀ ε > 0, ∃ s : Set X, s.Countable ∧ ∀ x, ∃ y ∈ s, dist x y ≤ ε := by
+    intro ε hpos
+    let U := fun (z : X) ↦ Metric.ball z ε
+    have hU : ∀ z, U z ∈ 𝓝 z := by
```

Comment:
I see. In principle, I very much agree to not delay every code motion by drive-by refactoring. In practice, Github's interface doesn't show code motion, so moved code often gets re-reviewed, for pre-existing issues :-(

How about this: would you like to make a PR to this code on `master`, with the changes I suggested? I can approve that one quickly, and then this PR can move the tweaked code. (I can also make that PR, but then I need to find another reviewer.)

Reply:
https://github.com/leanprover-community/mathlib4/pull/13515

Reply:
Oh I just realized I did the move in that PR rather than just fixing the issues you pointed out; hopefully that's not a big deal.

Reply:
Well, the problem is that when reviewing this PR it is not clear whether the changes are already taken care of in the other, so it is better to avoid duplicating the changes.